### PR TITLE
Upgrade Merlin to OCaml 4.14

### PR DESCRIPTION
**Pull request by voodoos**
## Roadmap

- [x] first draft which compiles succesfully
- [x] fix testsuite 
- [ ] merge new locate code when ready

@kit-ty-kate do you know if it is already possible to have some CI (Github or Ocaml CI) working here ?

### CODE REVIEW

**Comment by kit-ty-kate** [comment]
We can’t enable 4.14 in OCaml-CI before the the shape fix is merged, otherwise it’ll break our cluster.
Github Actions should be fine. I think changing this [line](https://github.com/ocaml/merlin/blob/master/.github/workflows/main.yml#L27) to `ocaml-variants.4.14.0+trunk` should be fine

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
           - ubuntu-latest
           - windows-latest
         ocaml-version:
```
Comment (kit-ty-kate) [inline .github/workflows/main.yml:26]:
```suggestion
        ocaml-compiler:
```

**Comment by voodoos** [comment]
CI is now running and failing with _panache_ ! Thank you @kit-ty-kate

**Comment by let-def** [comment]
I have reviewed this branch and I believe it is good. I made minor changes after rebasing on top of Merlin master and OCaml 4.14 trunk.
I approve the changes, though I will test it for a few more days.

**Review by let-def — APPROVED** [review]

**Comment by trefis** [comment]
I've also been using this for a few weeks and it works fine, but I noticed some changes with the upstream version today.
https://github.com/ocaml/ocaml/pull/10488 in particular doesn't seem to exist in Merlin. Was that a deliberate choice?

**Comment by voodoos** [comment]
@trefis I've redone the merge of `Printtyp`. The test suite allowed me to find a small printing bug in the compiler (see [e273cc7](https://github.com/ocaml/merlin/pull/1415/commits/e273cc718fbc5dc6f3d106870162e63c51dff6af)).

**Comment by nojb** [comment]
Hiya. Is this the official 4.14 branch? I need to adapt some inhouse Merlin code and would not want to base it on the wrong one :)

**Comment by nojb** [comment]
> Yes this is the 414 branch that will be merged soon into master for the official release. It is up-to-date with the recent release of OCaml 4.14.

Thanks!

**Review by trefis — APPROVED** [review]
LGTM.

I've also been using this branch for weeks (if not months) now, and it seems OK.

**Pull request merged**